flight/PiOS/STM32F4xx/pios_fskdac.c

```diff
@@ -0,0 +1,352 @@
+	uint8_t cur_byte;
+	uint8_t cur_parity;
+
+	//! The phase when starting and ending this symbol
+	float starting_phase;
+	float ending_phase;
+
+	pios_com_callback tx_out_cb;
+	uintptr_t tx_out_context;
+};
+
+// Matlab code to generate symbols
+/*
+DAC_SBS = 50000; F1 = 1200; p1 = 2; F2 = 2200; p2 = 3; % space period of each to allow allignment
+mark = round((-sin(linspace(0,2*pi*p1,DAC_SBS/F1*p1))*(2^12/3.3*0.5)+(2^12/3.3*0.5)));
+space = round((-sin(linspace(0,2*pi*p2,DAC_SBS/F2*p2))*(2^12/3.3*0.5)+(2^12/3.3*0.5)));
+blank = zeros(1,round(DAC_SBS/1200)) + round((2^12/3.3*0.5));
+['const uint16_t MARK_SAMPLES[] = {', sprintf('%d,', mark), '};']
+['const uint16_t SPACE_SAMPLES[] = {', sprintf('%d,', space), '};']
+['const uint16_t BLANK_SAMPLES[] = {', sprintf('%d,', blank), '};']
+sprintf('SAMPLES_PER_BIT = %d ',round(DAC_SBS/1200)) % duration of symbol at 1200bps
+sprintf('SAMPLES_PER_BIT = %d ',DAC_SBS) 
+*/
+
+const uint16_t MARK_SAMPLES[] = {621,573,527,483,442,405,374,348,329,316,311,312,321,337,360,389,423,462,505,550,597,644,691,737,779,818,852,881,904,920,929,931,925,913,893,867,836,799,758,714,668,621,573,527,483,442,405,374,348,329,316,311,312,321,337,360,389,423,462,505,550,597,644,691,737,779,818,852,881,904,920,929,931,925,913,893,867,836,799,758,714,668,621,};
+const uint16_t SPACE_SAMPLES[] = {621,534,455,389,341,314,312,335,379,443,521,606,693,774,843,894,924,930,912,871,810,734,650,563,480,409,354,321,310,325,362,420,494,577,664,748,821,879,917,931,921,887,832,761,678,592,507,431,370,329,311,317,347,399,468,549,635,721,798,862,907,929,927,901,852,786,707,621,};
+const uint16_t BLANK_SAMPLES[] = {621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,621,};
+static const uint32_t SAMPLES_PER_BIT = 42;
+static const uint32_t DAC_SBS = 50000;
+const float MARK_SAMPLES_PER_CYCLE = 41.666667f;
+const float SPACE_SAMPLES_PER_CYCLE = 22.727273f;
+
+static bool PIOS_FSKDAC_validate(struct pios_fskdac_dev * fskdac_dev)
+{
+	return (fskdac_dev->magic == PIOS_FSKDAC_DEV_MAGIC);
+}
+
+static struct pios_fskdac_dev * PIOS_FSKDAC_alloc(void)
+{
+	struct pios_fskdac_dev * fskdac_dev;
+
+	fskdac_dev = (struct pios_fskdac_dev *)PIOS_malloc(sizeof(*fskdac_dev));
+	if (!fskdac_dev) return(NULL);
+
+	memset(fskdac_dev, 0, sizeof(*fskdac_dev));
+	fskdac_dev->magic = PIOS_FSKDAC_DEV_MAGIC;
+	return(fskdac_dev);
+}
+
+struct pios_fskdac_dev * g_fskdac_dev;
+
+/**
+* Initialise a single USART device
+*/
+int32_t PIOS_FSKDAC_Init(uintptr_t * fskdac_id)
+{
+	PIOS_DEBUG_Assert(fskdac_id);
+	PIOS_DEBUG_Assert(cfg);
+
+	struct pios_fskdac_dev * fskdac_dev;
+
+	fskdac_dev = (struct pios_fskdac_dev *) PIOS_FSKDAC_alloc();
+	if (!fskdac_dev) return -1;
+
+	// Handle for the IRQ
+	g_fskdac_dev = fskdac_dev;
+
+
+	PIOS_DAC_COMMON_Init(PIOS_FSKDAC_DMA_irq_cb);
+
+	TIM_SetAutoreload(TIM6, PIOS_PERIPHERAL_APB1_CLOCK / DAC_SBS);
```

Comment:
not if they correctly define that frequency, this should make it do the right thing